CustomHeaderSettingsForm.inc.php

```diff
@@ -73,8 +76,25 @@ function execute(...$functionArgs) {
 
 		$request = Application::get()->getRequest();
 		$this->_plugin->updateSetting($this->_contextId, 'content', $this->getData('content'), 'string');
+		$this->_plugin->updateSetting($this->_contextId, 'backendContent', $this->getData('backendContent'), 'string');
 		$this->_plugin->updateSetting($this->_contextId, 'footerContent', $this->getData('footerContent'), 'string');
 		$notificationManager = new NotificationManager();
 		$notificationManager->createTrivialNotification($request->getUser()->getId(), NOTIFICATION_TYPE_SUCCESS);
 	}
+
+	/**
+	 * Validate that the input is well-formed XML
+	 * We want to avoid breaking the whole HTML page with an unclosed HTML attribute quote or tag
+	 * @param $input string
+	 * @return boolean
+	 */
+	function validateWellFormed($input) {
+		$libxml_errors_setting = libxml_use_internal_errors();
+		libxml_use_internal_errors(true);
+		libxml_clear_errors();
+		$xml = simplexml_load_string($input);
```

Comment:
Just a thought -- is this going to be overly restrictive? It's requiring well-formed XML, but the field should be checking for well-formed HTML. I think the DOM tools might provide an alternative:

https://www.php.net/manual/en/domdocument.loadhtml.php

The idea would be:

1. Use the same `libxml_use_internal_errors` approach and return value to determine whether a fatal parsing error occurred
2. If it's flexible enough to allow parsing of documents that we don't want to permit, maybe re-serialize the HTML before storing in the setting to resolve that?

Reply:
Fair enough.  It's unlikely that someone would want to legitimately throw an unclosed `li` in the `head`, but I could see someone wanting to do an old-style `meta` tag perhaps.

Reply:
@ctgraham, unfortunately it looks like this won't work well. See:

https://github.com/pkp/customHeader/issues/33

I'll follow up on that issue.

Reply:
Perhaps and additional checkbox or a second confirmation "submit" which bypasses the validation? A "I solemnly swear I am up to no good" prompt?

Reply:
@ctgraham, that sounds OK to me (and I'm not sure we'll find a better solution). I did find an OK-looking work-around that gets the validator to pass without breaking the JS, which I described [here](https://github.com/pkp/customHeader/issues/33#issuecomment-1868090661).

Reply:
"I solemnly swear I am up to no good" 🧙‍♂️ totally yes! After all who doesn't know that "with great power comes great responsibility"?